diff --git a/dist/cjs/components/Modal/Modal.js b/dist/cjs/components/Modal/Modal.js
index 7741145310e15b3dd90a16d4a3d02e8655dac801..6cbe36d8d065db6f7ff9bc65359994c4281c8010 100644
--- a/dist/cjs/components/Modal/Modal.js
+++ b/dist/cjs/components/Modal/Modal.js
@@ -25,7 +25,7 @@ const Modal = exports.Modal = (0, _utils.typedMemo)(props => {
   }, []);
   (0, _react.useEffect)(() => {
     return () => {
-      if (modalContainer) {
+      if (modalContainer && modalContainer.parentNode === document.body) {
         document.body.removeChild(modalContainer);
       }
     };
diff --git a/dist/es/components/Modal/Modal.js b/dist/es/components/Modal/Modal.js
index 69ccb918301fd815994d53e9f45c768dcb950bc2..92922b9469c39c20ea80387d069ca243b9bcb7af 100644
--- a/dist/es/components/Modal/Modal.js
+++ b/dist/es/components/Modal/Modal.js
@@ -17,7 +17,7 @@ export const Modal = typedMemo(props => {
   }, []);
   useEffect(() => {
     return () => {
-      if (modalContainer) {
+      if (modalContainer && modalContainer.parentNode === document.body) {
         document.body.removeChild(modalContainer);
       }
     };
